sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -115,9 +113,27 @@ class SQLContext private[sql](
   }
 
   /**
-   * @return Spark SQL configuration
+   * Per-session state, e.g. configuration, functions, temporary tables etc.
    */
-  protected[sql] lazy val conf = new SQLConf
+  @transient
+  protected[sql] lazy val sessionState: SessionState = new SessionState(self)
+  protected[sql] lazy val conf: SQLConf = sessionState.conf
```

Comment:
Yea def is better - just reducing the number of fields and you also don't need to make them transients.